src/client.js

```diff
@@ -148,15 +145,31 @@ assign(Client.prototype, {
 
   Pool: Pool2,
 
-  initializePool: function(config) {
-    if (this.pool) this.destroy()
-    this.pool = new this.Pool(assign(this.poolDefaults(config.pool || {}), config.pool))
-    this.pool.on('error', function(err) {
-      helpers.error('Pool2 - ' + err)
-    })
-    this.pool.on('warn', function(msg) {
-      helpers.warn('Pool2 - ' + msg)
+  initializePool: function(config, callback) {
+    var client = this
+    var promise = Promise.try(function() {
+      return Promise.try(function() {
+        if (client.pool) {
+          return client.destroy()
+        }
+      })
+      .then(function() {
+        client.pool = new client.Pool(assign(client.poolDefaults(config.pool || {}), config.pool))
+        client.pool.setMaxListeners(0) // Suppress warnings about too many event listeners
```

Comment:
@rhys-vdw because I added per-connection listeners in [`client#releaseConnection`](https://github.com/tgriesser/knex/pull/1014/files#diff-50cfa59973c04321b5da0c6da0fdf4feR249) and it would spam warnings about having more than 10 event listeners whenever you have several connections open. Maybe @myndzi (from [Pool2](https://github.com/myndzi/pool2)) has some input here on whether there is another way to wait for connections to be released.